yarn-project/circuit-types/src/logs/unencrypted_l2_log.ts

```diff
@@ -24,6 +24,7 @@ export class UnencryptedL2Log {
   ) {}
 
   get length(): number {
+    // TODO(6578): explain magic number 4 here
```

Comment:
This +4 exists because when we serialise:
```
  public toBuffer(): Buffer {
    return Buffer.concat([
      this.contractAddress.toBuffer(),
      this.selector.toBuffer(),
      prefixBufferWithLength(this.data), <--- prefixing with length
    ]);
  }
```
We prefix the data portion with its own length in 4 bytes. This is useful as the data is variable length so we can tell readers how to process it.
I don't think it's strictly necessary though, because the reader should get the full length of this buffer, read the first 32 for the address, read the next 4 for the event selector, then simply read the rest. This is how it works for encrypted event logs, for example.
(Hope this helps - also pasting this comment in the issue!)